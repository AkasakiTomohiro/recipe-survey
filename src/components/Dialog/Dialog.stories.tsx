/**
 * @file Dialog.stories.tsx
 * @author Artan's Projects
 * @copyright © 2024 Artan's Projects. All rights reserved.
 */

import { Dialog } from './Dialog';

import type { Meta, StoryObj } from '@storybook/react';

const meta: Meta<typeof Dialog> = {
  title : 'RecipeSurvey/components/Dialog',
  render: (args): JSX.Element => <Dialog {...args} />
};

export default meta;

type Story = StoryObj<typeof Dialog>;

export const Primary: Story = {
  args: {
    open       : true,
    title      : 'タイトル',
    submitLabel: 'ボタン名',
    children   : (<>コンテンツ</>),
    onClose    : () => console.log('cancel'),
    onSubmit   : async () => console.log('submit')
  }
};

export const Cancel: Story = {
  args: {
    open       : true,
    title      : 'タイトル',
    submitLabel: 'ボタン名',
    cancelLabel: 'キャンセル',
    children   : (<>コンテンツ</>),
    onClose    : () => console.log('cancel'),
    onSubmit   : async () => console.log('submit')
  }
};
