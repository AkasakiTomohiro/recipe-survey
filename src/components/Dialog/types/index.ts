/**
 * @file index.ts
 * @author Artan's Projects
 * @copyright © 2024 Artan's Projects. All rights reserved.
 */

import { Theme } from '@mui/material';
import { SxProps } from '@mui/system';
import { Breakpoint } from '@mui/system/createTheme/createBreakpoints';

export interface IDialogProps {

  /**
   * タイトル
   */
  title: string;

  /**
   * 確定ボタンのラベル
   */
  submitLabel: string;

  /**
   * キャンセルボタンのラベル
   */
  cancelLabel?: string;

  /**
   * モーダルの表示／非表示制御
   */
  open: boolean;

  /**
   * モーダルのcloseイベント
   */
  onClose: () => void;

  /**
   * 確定ボタン押下イベント
   */
  onSubmit: () => void;

  /**
   * DialogContentの中の要素
   */
  children: React.ReactNode;

  /**
  * ダイアログの最大幅
  */
  maxWidth?: Breakpoint | false;

  /**
   * CSS
   */
  sx?: SxProps<Theme>;
}