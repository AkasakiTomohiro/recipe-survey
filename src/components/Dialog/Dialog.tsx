/**
 * @file Dialog.tsx
 * @author Artan's Projects
 * @copyright Â© 2024 Artan's Projects. All rights reserved.
 */

import { IconButton } from '@mui/material';
import Button from '@mui/material/Button';
import MuiDialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogContent from '@mui/material/DialogContent';
import DialogTitle from '@mui/material/DialogTitle';
import { styled, Theme } from '@mui/material/styles';

import { CloseIcon } from '../Icon';
import { IDialogProps } from './types';

const CustomIconButton = styled(IconButton)(({ theme }) => ({
  position: 'absolute',
  right   : theme.spacing(1),
  top     : theme.spacing(1),
  color   : theme.palette.grey[500]
}));

export const Dialog = (props: IDialogProps): JSX.Element => {
  return (
    <MuiDialog
      sx={props.sx}
      open={props.open}
      fullWidth={true}
      maxWidth={props.maxWidth}
      scroll="paper"
      aria-labelledby="scroll-dialog-title"
      aria-describedby="scroll-dialog-description"
    >
    <DialogTitle>
      {props.title}
      <CustomIconButton aria-label="close" onClick={props.onClose}>
        <CloseIcon />
      </CustomIconButton>
    </DialogTitle>
    <DialogContent dividers>
      {props.children}
    </DialogContent>
    <DialogActions sx={{ justifyContent: props.cancelLabel && 'space-between' }}>
      <Button variant="contained" color="secondary" onClick={props.onSubmit}>
        {props.submitLabel}
      </Button>
      {props.cancelLabel &&
      <Button variant="contained" color="secondary" onClick={props.onClose}>
        {props.cancelLabel}
      </Button>
      }
    </DialogActions>
  </MuiDialog>
  );
}