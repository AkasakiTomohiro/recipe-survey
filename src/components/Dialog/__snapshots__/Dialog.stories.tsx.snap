// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Dialog Cancel smoke-test 1`] = `""`;

exports[`components/Dialog Primary smoke-test 1`] = `""`;
